pkg/controller/storagecluster/initialization_reconciler.go

```diff
@@ -61,6 +70,103 @@ func (r *ReconcileStorageCluster) ensureStorageClasses(instance *ocsv1.StorageCl
 	return nil
 }
 
+// ensureExternalClusterConfig ensures that Storagclass for the external cluster exist
+func (r *ReconcileStorageCluster) ensureExternalClusterConfig(instance *ocsv1.StorageCluster, reqLogger logr.Logger) error {
+	// check for the status boolean value accepted or not
+	if instance.Status.ExternalSecretFound {
+		return nil
+	}
+	found := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "rook-ceph-external-cluster-details",
+			Namespace: "openshift-storage",
+		},
+
+		Data: map[string][]byte{
+			"external_cluster_details": []byte(""),
+		},
+	}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: found.Name, Namespace: found.Namespace}, found)
+	if err != nil { // external cluster already found
+		return err
+	}
+	ownerRef := metav1.OwnerReference{
+		UID:        instance.UID,
+		APIVersion: instance.APIVersion,
+		Kind:       instance.Kind,
+		Name:       instance.Name,
+	}
+	var data ExternalResources
+	err = json.Unmarshal([]byte(found.Data["external_cluster_details"]), &data)
+	if err != nil {
+		return fmt.Errorf("could not parse json blob, got %+v", err)
+	}
+	for _, d := range data {
+		switch d.Kind {
+		case "ConfigMap":
+			cm := &corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:            d.Name,
+					Namespace:       d.Name,
+					OwnerReferences: []metav1.OwnerReference{ownerRef},
+				},
+			}
+			err = controllerutil.SetControllerReference(instance, cm, r.scheme)
+			if err != nil {
+				return err
+			}
+			_, err = controllerutil.CreateOrUpdate(context.TODO(), r.client, cm, func() error {
```

Comment:
What about just using `r.client.Create`? We don't want to update anything here, also the check `!errors.IsAlreadyExists(err)` later will make more sense.

Reply:
What about just using `r.client.Create`? We don't want to update anything here,.
> Do you mean like this?. `_,err = r.client.Create(context.TODO(),s)`

 also the check `!errors.IsAlreadyExists(err)` later will make more sense.
>Do you meant that to check for `!errors.IsAlreadyExists(err) at the very end of the switch cases?.

Reply:
Yes and keep the check `!errors.IsAlreadyExists(err)` intact.